### PR TITLE
autoupdater: honor PR review status

### LINKED ISSUE #26: autoupdater: check during update step if PR is approved
If a PR is enqueued for updates but the number of required approvals is not met, it will block the queue until it gets stale.
The github rest api does not allow to query the approval status of a PR. 

Check if a PR is approved in the update step, before it gets updated.
If it isn't, suspend it.
If an approval event is received, check if updates for it should be resumed.

Via the REST API it is only possible to get the review state via a search query when search like:
`'https://api.github.com/search/issues?q=head:metrics+repo:simplesurance/goordinator+review=approved+is:pr'

Via graphQL it's possible to query the review state for a specific pr.

### PR DESCRIPTION
**Pull request by fho**
The autoupdater was ignoring the review state of pull-requests.
This caused that when:
- all status checks for a PR were successful,
- it was the first element in the queue,
- and review approval was configured as merge requirement in GitHub

that the PR got stuck at the first place, until the stale timeout was reached
and the PR was suspended.
Github was not automerging the PR because the approvals were missing.

When evaluating the status of a PR the approval status is now queried from the API. If the PR is not approved it is suspended.
If a review was submitted and the PR status is approved autoupdating is resumed.
If a dismissed event for a review is received and the PR is the first one in the
queue, it's approval status is reevaluated.
If a negative review (change request) event or a dismissed event for a review is
received, the approval status is checked again if the PR is first in the queue.

This change means that PRs can be enqueued for autoupdating despite their review
status is not approved, without blocking the queue unnecessarily.

Because the REST search API returned old/cached values for the combined approved status the graphQL API is queried instead.
Via the GraphQL endpoint I encountered recently the same issue but now I can not
reproduce it anymore.
It is not clear if it can also return a cached status or if that was a temporary
issue on GitHub.
If the GraphQL API has the same issue, possible workaround could be:
- get the updatedAt field via graphql for the last changed pull request review
  when querying the approval status and check that it is later then the time the
  last review even was received for the PR,
  if it's not later, query the approval status again,
- add a configuration option to manually configure the number of required review
  per repository, get from the API the number of review and if a PR has X
  approvals and no changes-requested reviews

Closes #26

### CODE REVIEW

File: internal/autoupdate/queue.go [hunk 1]
```diff
@@ -473,6 +473,33 @@ func (q *queue) updatePR(ctx context.Context, pr *PullRequest) {
...
+			)
+			return
+		}
+		logger.Info(
+			"updates suspended, pr is not approved",
+			logFieldReason("pr_not_approved"),
+			logEventUpdatesSuspended,
+			zap.Error(err),
+		)
+	}
```
Comment (FranciscoKurpiel) [inline internal/autoupdate/queue.go:501]: Shouldn't it return here even if `Suspend()` is not an error?

Comment (fho) [reply internal/autoupdate/queue.go:501]: definitely, nice catch, thanks!

Comment (fho) [reply internal/autoupdate/queue.go:504]: fixed

**Review by FranciscoKurpiel — APPROVED** [review]

File: internal/autoupdate/queue_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	"github.com/golang/mock/gomock"
+	"github.com/simplesurance/goordinator/internal/autoupdate/mocks"
+	"github.com/simplesurance/goordinator/internal/goordinator"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestUpdatePR_DoesNotCallBaseBranchUpdateIfPRIsNotApproved(t *testing.T) {
```
Comment (FranciscoKurpiel) [inline internal/autoupdate/queue_test.go:16]: :1st_place_medal:

**Pull request merged**